### DISCUSSION TITLE
cannot create 'data': too many devices in one vdev

### DISCUSSION THREAD

**Discussion by nnovikov** [openzfs/zfs#12325] [Q&A] [answered]
what is the maximum number of devices possible?

**Comment by rincebrain** [comment, marked as answer]
I don't know, offhand, but if you share the precise command you were using, we can try to predict whether this is expected or a poor error message trying to indicate something else and failing.

**Reply by nnovikov** [reply]
zpool create data draid  [240 disk here]

**Reply by behlendorf** [reply]
Each top-level dRAID vdev can support up to 255 devices.

**Reply by nnovikov** [reply]
Very strange. Why am I getting an error on 240 disks?

**Reply by behlendorf** [reply]
Can you share the exact command, the following works as expected for me:

```sh
$ truncate -s 64M /tmp/file.{01..255}
$ sudo zpool create tank draid /tmp/file.{01..255}

$ zpool status -v
  pool: tank
 state: ONLINE
config:

        NAME                   STATE     READ WRITE CKSUM
        tank                   ONLINE       0     0     0
          draid1:8d:255c:0s-0  ONLINE       0     0     0
            /tmp/file.001      ONLINE       0     0     0
            ...
```

**Reply by nnovikov** [reply]
May be some disks are broken, but I haven't any message about this in error.
My variant:

`zpool create data draid /dev/sdc /dev/sdd /dev/sde /dev/sdf /dev/sdg /dev/sdh /dev/sdi /dev/sdj /dev/sdk /dev/sdl /dev/sdm /dev/sdn /dev/sdo /dev/sdp /dev/sdq /dev/sdr /dev/sds /dev/sdt /dev/sdu /dev/sdv /dev/sdw /dev/sdx /dev/sdy /dev/sdz /dev/sdaa /dev/sdab /dev/sdac /dev/sdad /dev/sdae /dev/sdaf /dev/sdag /dev/sdah /dev/sdai /dev/sdaj /dev/sdak /dev/sdal /dev/sdam /dev/sdan /dev/sdao /dev/sdap /dev/sdaq /dev/sdar /dev/sdas /dev/sdat /dev/sdau /dev/sdav /dev/sdaw /dev/sdax /dev/sday /dev/sdaz /dev/sdba /dev/sdbb /dev/sdbc /dev/sdbd /dev/sdbe /dev/sdbf /dev/sdbg /dev/sdbh /dev/sdbi /dev/sdbj /dev/sdbk /dev/sdbl /dev/sdbm /dev/sdbn /dev/sdbo /dev/sdbp /dev/sdbq /dev/sdbr /dev/sdbs /dev/sdbt /dev/sdbu /dev/sdbv /dev/sdbw /dev/sdbx /dev/sdby /dev/sdbz /dev/sdca /dev/sdcb /dev/sdcc /dev/sdcd /dev/sdce /dev/sdcf /dev/sdcg /dev/sdch /dev/sdci /dev/sdcj /dev/sdck /dev/sdcl /dev/sdcm /dev/sdcn /dev/sdco /dev/sdcp /dev/sdcq /dev/sdcr /dev/sdcs /dev/sdct /dev/sdcu /dev/sdcv /dev/sdcw /dev/sdcx /dev/sdcy /dev/sdcz /dev/sdda /dev/sddb /dev/sddc /dev/sddd /dev/sdde /dev/sddf /dev/sddg /dev/sddh /dev/sddi /dev/sddj /dev/sddk /dev/sddl /dev/sddm /dev/sddn /dev/sddo /dev/sddp /dev/sddq /dev/sddr /dev/sdds /dev/sddt /dev/sddu /dev/sddv /dev/sddw /dev/sddx /dev/sddy /dev/sddz /dev/sdea /dev/sdeb /dev/sdec /dev/sded /dev/sdee /dev/sdef /dev/sdeg /dev/sdeh /dev/sdei /dev/sdej /dev/sdek /dev/sdel /dev/sdem /dev/sden /dev/sdeo /dev/sdep /dev/sdeq /dev/sder /dev/sdes /dev/sdet /dev/sdeu /dev/sdev /dev/sdew /dev/sdex /dev/sdey /dev/sdez /dev/sdfa /dev/sdfb /dev/sdfc /dev/sdfd /dev/sdfe /dev/sdff /dev/sdfg /dev/sdfh /dev/sdfi /dev/sdfj /dev/sdfk /dev/sdfl /dev/sdfm /dev/sdfn /dev/sdfo /dev/sdfp /dev/sdfq /dev/sdfr /dev/sdfs /dev/sdft /dev/sdfu /dev/sdfv /dev/sdfw /dev/sdfx /dev/sdfy /dev/sdfz /dev/sdga /dev/sdgb /dev/sdgc /dev/sdgd /dev/sdge /dev/sdgf /dev/sdgg /dev/sdgh /dev/sdgi /dev/sdgj /dev/sdgk /dev/sdgl /dev/sdgm /dev/sdgn /dev/sdgo /dev/sdgp /dev/sdgq /dev/sdgr /dev/sdgs /dev/sdgt /dev/sdgu /dev/sdgv /dev/sdgw /dev/sdgx /dev/sdgy /dev/sdgz /dev/sdha /dev/sdhb /dev/sdhc /dev/sdhd /dev/sdhe /dev/sdhf /dev/sdhg /dev/sdhh /dev/sdhi /dev/sdhj /dev/sdhk /dev/sdhl /dev/sdhm /dev/sdhn /dev/sdho /dev/sdhp /dev/sdhq /dev/sdhr /dev/sdhs /dev/sdht /dev/sdhu /dev/sdhv /dev/sdhw /dev/sdhx /dev/sdhy /dev/sdhz /dev/sdia /dev/sdib /dev/sdic /dev/sdid /dev/sdie /dev/sdif /dev/sdig /dev/sdih`

**Reply by pashford** [reply]
> Each top-level dRAID vdev can support up to 255 devices.

Is there a page of documentation containing this sort of limit?  If not, perhaps one should be created (zfs-limits?).

**Reply by behlendorf** [reply]
An explanation of the behavior of draid is available in the `zpoolconcepts(7)` [man page](https://openzfs.github.io/openzfs-docs/man/7/zpoolconcepts.7.html) however it doesn't include the maximum size (255).  I'd suggest we add it there.
